### ISSUE TITLE
Exchange NDR messages not signed

### ISSUE THREAD

**Issue by Besdima** [Pro/dkim-exchange#171]
Versions

Windows Server Version: Windows Server 2012 R2 Standard
Exchange Version: Exchange 2013 CU15 (15.0.1263.5)
Installed DKIM Exchange Version: 3.0.8

Description

Dkim signer didn't sing NDR messages from postmaster@domain.com, so gmail reject this messages with DMARC policy error: 550 5.7.1 This message was not accepted due to domain owner DMARC policy

In debug:
DEBUG: Got new message, checking if I can sign it...
DEBUG: Message is a System message or of TNEF format. Not signing.

**Comment by Pro** [comment]
I don't think that we can sign system messages (i.e., NDR messages), since they are read-only to the dkim signer:

https://github.com/Pro/dkim-exchange/blob/master/Src/Exchange.DkimSigner/DkimSigningRoutingAgent.cs#L94

I need to investigate that further.

Btw, the error message says "DMARC policy" problem, which is not the same as DKIM. Not sure if the error message is just not clear enough or the mail is rejected due to some other config.

**Comment by Besdima** [comment]
Hi!
I have look at mail headers in NDR messages, and it wasn't DKIM sign in there.

**Comment by RickStinson** [comment]
+1 for this issue.
Rejected Postmaster Mails are definitly a problem - espacially when people gets no "undelivered" messages back.

Workaround: use as postmaster address an subdomain without DKIM (eg postmaster@subdomain.mydomain.com)

**Comment by lucacervo** [comment]
Good Afternoon,
i have the same problem with NDR Messages. I have tried this configuration:
TNEF -> Disabled; Postmaster Address: member of a subdomain of my primary mail domain; 
Anytime my exchange server delivers an NDR i see in the event log this message:
**_DEBUG: Message is a  System message or of TNEF format. Not signing._**

Exchange Version: Version 15.0 (Build 1210.3)
Windows Server 2012 R2
DKIM Signer version 3.2.8

The only way to send NDR correctly is to use a completely different domain. I suspect that DKimSigner signs tries to sign also the subdomain ones.

Can you please give me support?
Thanks, best regards

**Comment by Deckard99** [comment]
Any news on this issue? Just stumbled across at an installation, that NDRs won't be signed. Using a postmasteraddress from a subdomain without DMARC is a weak workaround I guess.

Regards

**Comment by FXEGIT** [comment]
Same issue here : https://github.com/Pro/dkim-exchange/issues/395.
